src/main/java/com/faforever/moderatorclient/ui/ViewHelper.java

```diff
@@ -1002,6 +1004,43 @@ public static void buildMapTreeView(TreeTableView<MapTableItemAdapter> mapTreeVi
         nameColumn.setMinWidth(300);
         mapTreeView.getColumns().add(nameColumn);
 
+        TreeTableColumn<MapTableItemAdapter, MapTableItemAdapter> favoriteColumn = new TreeTableColumn<>("Favorite");
+        favoriteColumn.setMinWidth(100);
+        favoriteColumn.setCellValueFactory(new TreeItemPropertyValueFactory<>("this"));
+        Callback<TreeTableColumn<MapTableItemAdapter, MapTableItemAdapter>, TreeTableCell<MapTableItemAdapter, MapTableItemAdapter>> cellFactory = new Callback<TreeTableColumn<MapTableItemAdapter, MapTableItemAdapter>, TreeTableCell<MapTableItemAdapter, MapTableItemAdapter>>() {
+            @Override
+            public TreeTableCell<MapTableItemAdapter, MapTableItemAdapter> call(TreeTableColumn<MapTableItemAdapter, MapTableItemAdapter> param) {
+                return new TreeTableCell<MapTableItemAdapter, MapTableItemAdapter>() {
+                    @Override
+                    public void updateItem(MapTableItemAdapter item, boolean empty) {
+                        super.updateItem(item, empty);
+                        if (!empty && item != null && !item.isMapVersion()) {
+                            if (ladderMapPoolController.isMapFavorite(Integer.parseInt(item.getId()))) {
+                                Button button = new Button("Unfavorite");
+                                button.setOnMouseClicked(event -> removeFavorite.accept(item));
+                                button.setTextFill(Color.rgb(200, 10, 10));
+
+                                setGraphic(button);
+                                return;
+                            } else {
+                                Button button = new Button("Favorite");
+                                button.setOnMouseClicked(event -> addFavorite.accept(item));
```

Comment:
nullability of addFavorite

Reply:
Done.

Reply:
What was changed here? I don't even know what @Sheikah45 refers to O.o

Reply:
addFavorite is marked as nullable in the method parameters But there is no check to see if it is nonNull before calling accept on it. @Spikey84 I meant checking for null on addFavorite and removeFavorite not the item